src/AMAMM.sol

```diff
@@ -71,32 +71,34 @@ contract AMAMM is IAmAmm {
     /// Bidder actions
     /// -----------------------------------------------------------------------
     /// @inheritdoc IAmAmm
-    function bid(PoolId id, bytes7 payload, uint128 rent, uint128 deposit, uint40 _epoch) external virtual override {
+    function bid(PoolId id, bytes7 payload, uint128 rent, uint40 _epoch) external virtual override {
         /// -----------------------------------------------------------------------
         /// Validation
         /// -----------------------------------------------------------------------
 
         address msgSender = LibMulticaller.senderOrSigner();
 
-        depositToken(id, msgSender, _epoch);
-        // ensure bid is valid
-        // - manager can't be zero address
-        // - bid needs to be greater than the next bid by >10%
-        // - deposit needs to cover the rent for K hours
-        // - deposit needs to be a multiple of rent
-        // - payload needs to be valid
+        uint128 deposit = depositToken(id, msgSender, _epoch);
+
         if (
-            rent <= poolEpochManager[id][_epoch].rent.mulWad(MIN_BID_MULTIPLIER(id)) || deposit < rent * K(id)
+            _epoch > _getEpoch(id, block.timestamp)
+                || rent <= poolEpochManager[id][_epoch].rent.mulWad(MIN_BID_MULTIPLIER(id)) || deposit < rent * K(id)
```

Comment:
`deposit < rent * K` here has a duplicate calculation in the `depositToken` function. Hence I reckon here could just check `userBalance > rent * K`